src/appleseed-max-impl/appleseedinteractive/appleseedinteractive.cpp

```diff
@@ -305,32 +376,40 @@ asf::auto_release_ptr<asr::Project> AppleseedInteractiveRender::prepare_project(
     return project;
 }
 
-void AppleseedInteractiveRender::update_camera_parameters(INode* camera)
+void AppleseedInteractiveRender::update_camera_object(INode* camera)
 {
     ViewParams view_params;
     get_view_params_from_view_node(view_params, camera, m_time);
 
-    asr::ParamArray& params = m_project->get_scene()->get_active_camera()->get_parameters();
-    params.insert("horizontal_fov", asf::rad_to_deg(view_params.fov));
-
-#if MAX_RELEASE >= 18000
-    MaxSDK::IPhysicalCamera* phys_camera = dynamic_cast<MaxSDK::IPhysicalCamera*>(camera->EvalWorldState(m_time).obj);
-
-    if (phys_camera && phys_camera->GetDOFEnabled(m_time, FOREVER))
-    {
-        set_camera_dof_params(params, phys_camera, m_bitmap, m_time);
-    }
-#endif
+    auto new_camera = build_camera(camera, view_params, m_bitmap, RendererSettings::defaults(), m_time);
+    get_render_session()->get_render_controller()->schedule_update(
```

Comment:
I would suggest getting rid of `get_render_session()` and using `m_render_session` instead. The less stuff we expose, the clearer the code.

Reply:
I think I would go even further: I would not expose the renderer controller via `get_render_controller()`, but instead move `schedule_update()` to the render session.

Reply:
Moved.